src/libraries/illink-sharedframework.targets

```diff
@@ -30,8 +30,20 @@
       <ILLinkArgs>$(ILLinkArgs) --nowarn $(LinkerNoWarn)</ILLinkArgs>
     </PropertyGroup>
 
-     <!-- Retrieve CoreLib's targetpath via GetTargetPath as it isn't binplaced yet. -->
-    <MSBuild Projects="$(CoreLibProject)"
+    <!-- Retrieve CoreLib's path as it isn't binplaced alongside the libraries. -->
+    <ItemGroup>
+      <CoreLibProjectWithRuntimeConfig Include="$(CoreLibProject)">
+        <!-- Don't flow TargetFramework and Platform to use same inputs and outputs as the CoreLib's build as part of the runtime. -->
+        <UndefineProperties>$(UndefineProperties);TargetFramework;Platform</UndefineProperties>
+        <!-- If conflicting, manually set the Configuration property of the CoreLib project so that it aligns with the specified RuntimeConfiguration in the libraries' build. -->
+        <Properties Condition="'$(RuntimeFlavor)' == 'CoreCLR' and
+                               '$(Configuration)' != '$(CoreCLRConfiguration)'">Configuration=$(CoreCLRConfiguration)</Properties>
```

Comment:
Just a question, mostly for @ViktorHofer. Why do we have both `$(CoreCLRConfiguration)` and `$(MonoConfiguration)`? Why not just have `$(RuntimeConfiguration)`? Do we expect to be able to build the repo using different configurations for coreclr and mono?

Reply:
There is a `RuntimeConfiguration` property but in this case we can't use that as we need to respect the inferred `CoreCLRConfiguration` and `MonoConfiguration` properties.

> Do we expect to be able to build the repo using different configurations for coreclr and mono?

Yes